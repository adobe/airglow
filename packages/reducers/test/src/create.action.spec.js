/*
Copyright 2019 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import createAction from '../../src/create.action';

const type = 'MY_ACTION';

describe('addFunctionProp', () => {
  beforeEach(() => {

  });
  it('creates an action', function () {
    const action = createAction(type);
    expect(action()).toMatchSnapshot();
  });
  it('passes in properties', function () {
    const action = createAction(type);
    expect(action({ a: 'b' })).toMatchSnapshot();
  });
  it('uses a provided payload mapper', function () {
    const mapper = (a, b) => ({ a, b });
    const action = createAction(type, mapper);
    expect(action(7, 23)).toMatchSnapshot();
  });
  it('uses a provided metdata mapper', function () {
    const mapper = (a, b) => ({ a, b });
    const action = createAction(type, null, mapper);
    expect(action(7, 23)).toMatchSnapshot();
  });
});
