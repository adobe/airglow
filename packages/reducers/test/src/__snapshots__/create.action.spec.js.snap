// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addFunctionProp creates an action 1`] = `
Object {
  "meta": undefined,
  "payload": undefined,
  "type": "MY_ACTION",
}
`;

exports[`addFunctionProp passes in properties 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "a": "b",
  },
  "type": "MY_ACTION",
}
`;

exports[`addFunctionProp uses a provided metdata mapper 1`] = `
Object {
  "meta": Object {
    "a": 7,
    "b": 23,
  },
  "payload": 7,
  "type": "MY_ACTION",
}
`;

exports[`addFunctionProp uses a provided payload mapper 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "a": 7,
    "b": 23,
  },
  "type": "MY_ACTION",
}
`;
